actor/src/main/scala/org/apache/pekko/dispatch/ThreadPoolBuilder.scala

```diff
@@ -85,6 +85,19 @@ final case class ThreadPoolConfig(
     queueFactory: ThreadPoolConfig.QueueFactory = ThreadPoolConfig.linkedBlockingQueue(),
     rejectionPolicy: RejectedExecutionHandler = ThreadPoolConfig.defaultRejectionPolicy)
     extends ExecutorServiceFactoryProvider {
+  // Written explicitly to permit non-inlined defn; this is necessary for downstream instrumentation that stores extra
+  // context information on the config
+  @noinline
+  def copy(
+      allowCorePoolTimeout: Boolean = allowCorePoolTimeout,
+      corePoolSize: Int = corePoolSize,
+      maxPoolSize: Int = maxPoolSize,
+      threadTimeout: Duration = threadTimeout,
+      queueFactory: ThreadPoolConfig.QueueFactory = queueFactory,
+      rejectionPolicy: RejectedExecutionHandler = rejectionPolicy
+  ): ThreadPoolConfig =
+    ThreadPoolConfig(allowCorePoolTimeout, corePoolSize, maxPoolSize, threadTimeout, queueFactory, rejectionPolicy)
```

Comment:
I spent a little bit of time on this yesterday and I got it wrong - I thought that this would only work if it was declared with `override` because in theory this function overrides the default `copy` function that gets added to case classes.

I still think it is quite odd to instrument the copy function as a way to instrument pinned dispatchers. I raised https://github.com/kamon-io/Kamon/issues/1366 about seeing if there is a better place to instrument this.

Reply:
Yeah this one left a slightly bad taste in my mouth and there probably is a better way. Ultimately I'd rather focus improvements on exposing deliberate instrumentation hooks from within pekko and changing the downstream to use them, because otherwise there's an inherent fragility... Any weaving approach can only work if there's a stable non-inlined API that's deliberately left for that purpose -- there's almost certainly places where we're currently only passing because of the whims of the scala compiler. However, such an approach will require considerably more time and thought. Since the Kamon library is probably one of the more widely used ways of instrumenting pekko, I think this is helpful as a stopgap that doesn't block people from adopting 1.1.x; but I completely understand your point of view here